kafka/src/test/java/org/apache/apex/malhar/kafka/KafkaInputOperatorTest.java

```diff
@@ -232,33 +228,11 @@ public void endWindow()
       int tupleSize = windowTupleCollector.size();
       tupleCollection.addAll(windowTupleCollector);
       
-      int countDownTupleSize = countDownAll ? tupleSize : endTuples;
+      int countDownTupleSize = endTuples;
 
       if (latch != null) {
-        Assert.assertTrue("received END_TUPLES more than expected.", latch.getCount() >= countDownTupleSize);
```

Comment:
No need to do double check here. The total number of received tuples are recorded and we can just do assertions there.